node.c

```diff
@@ -0,0 +1,74 @@
+#include "node.h"
+#include "board.h"
+
+
+void initializeRoot(node* root, struct board *boardState){
+
+    // Setting default values
+    root->parent = NULL;
+    root->childMoves = (int *)malloc(2 * sizeof(int));
+    root->numChildren = 0;
+    root->numWins = 0;
+    root->board = *boardState;
+    root->numSimulations = 0;
+    root->children = (node **)malloc(sizeof(node*));
+}
+
+
+node* expand(node* parentNode, int* action, int name){
+
+    // Dynamically allocate new struct node
+    node* newNode = (node *)malloc(sizeof(node));
+
+    // Setting member variables for new node
+    newNode->parent = parentNode;
+    newNode->childMoves = (int *)malloc(2 * sizeof(int));
+    newNode->numChildren = 0;
+    newNode->numWins = 0;
+    newNode->numSimulations = 0;
+    newNode->children = (node **)malloc(sizeof(node*));
+
+    copyBoard(&parentNode->board, &newNode->board);
+    placePiece(&newNode->board, action);
+
+
+    // Update parent node
+    parentNode->children = (node **)realloc(parentNode->children, (parentNode->numChildren + 1) * sizeof(node*));
```

Comment:
You should reallocate this array by doubling its size. Once you fill the array, you should realloc and double the size again. You should be able to check this if the number of children is divisible by two.

 - [ ] Update size of parent node children array by doubling the size rather than increasing the size by 1 every time